### PR TITLE
soc: intel_ish: Make ISH support APIC timer with TSC time source.

### PR DESCRIPTION
**Pull request by kwd-doodling**
This commit enables ISH boards to use APIC timer with TSC time source as
their system timer by replacing CONFIG_HPET with CONFIG_APIC_TIMER_TSC.

### CODE REVIEW

**Review by teburd — DISMISSED** [review]

**Comment by teburd** [comment]
I guess a question here would be, as @andyross noted, why don't we use the apic_tsc timer driver in place of apic, if the end goal is using apic tsc?

**Comment by kwd-doodling** [comment]
> I guess a question here would be, as @andyross noted, why don't we use the apic_tsc timer driver in place of apic, if the end goal is using apic tsc with #72839

ISH is a legacy CPU in fact :) Run apic_tsc on ISH already, ASSERT on "__ASSERT((ecx & BIT(24)) != 0, "No TSC Deadline support");"

**Review by teburd — DISMISSED** [review]

**Comment by kwd-doodling** [comment]
update this patch after https://github.com/zephyrproject-rtos/zephyr/pull/72839 was abandoned and https://github.com/zephyrproject-rtos/zephyr/pull/73185 was merged.

**Review by teburd — DISMISSED** [review]

File: soc/intel/intel_ish/Kconfig [hunk 1]
```diff
@@ -13,3 +13,4 @@ config SOC_FAMILY_INTEL_ISH
 	select INTEL_HAL
 	select HAS_PM
 	select HAS_COVERAGE_SUPPORT
+	select DYNAMIC_INTERRUPTS
```
Comment (edersondisouza) [inline soc/intel/intel_ish/Kconfig:16]: This seems unrelated - is this really needed?

Comment (kwd-doodling) [reply soc/intel/intel_ish/Kconfig:16]: It's required by APIC timer driver drivers/timer/apic_tsc.c who calls irq_connect_dynamic() function.

Comment (edersondisouza) [reply soc/intel/intel_ish/Kconfig:16]: Not sure how is the normal practice here, but shouldn't it be defined at `drivers/timer/Kconfig.x86` then?

Comment (kwd-doodling) [reply soc/intel/intel_ish/Kconfig]: all existing x86 CPUs select DYNAMIC_INTERRUPTS in SoC Kconfig, so I just follow. Maybe they have other reason to select, not only for APIC timer driver.
I uploaded a new patch to add dependency of DYNAMIC_INTERRUPTS for APIC TSC timer, https://github.com/zephyrproject-rtos/zephyr/pull/74127.
In addition, for ISH CPUs, select DYNAMIC_INTERRUPTS only when APIC_TIMER_TSC is set.

Comment (edersondisouza) [reply soc/intel/intel_ish/Kconfig]: I guess that making it clear on the driver that it depends on the dynamic interrupts is useful to figure out the reason for the selection here. On a side note, looking into dynamic interrupts and x86, things seem a bit messy - guess I'll add some things on my TODO list...

**Review by edersondisouza — APPROVED** [review]

**Review by teburd — APPROVED** [review]
